src/util.js

```diff
@@ -82,4 +82,46 @@ export function phrase (me, id, ...args) {
 	// We're on the root class and still can't find it
 	// Fall back to just displaying the id and all the args right after it
 	return id + " " + args.join(" ");
+}
+
+/**
+ * Test whether a URL matches a set of criteria
+ * @param {string | URL} url
+ * @param {{protocol, host, urls} | {protocol, host, urls}[]} criteria
+ * @returns {boolean}
+ */
+export function testURL (url, criteria) {
+	if (typeof url === "string") {
+		url = new URL(url);
+	}
+
+	if (Array.isArray(criteria.urls)) {
```

Comment:
```suggestion
	if (Array.isArray(criteria)) {
```

Reply:
When we have `urls`, we pass them as `criteria` in the `test()` method. So, we already have the needed array here.